### PR TITLE
add total energy 3d bubble test

### PR DESCRIPTION
**Pull request by szy21**
co-authored-by: @akshaysridhar 

Still needs:
- [x] Check conservation
- [x] Make some plots to see whether it is correct
- [x] hook up CI

### CODE REVIEW

File: examples/hybrid/bubble3d_invariant_totalenergy.jl [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+    χe = @. dρe = hwdiv(hgrad(cρe / cρ)) # we store χe in dρe
+    χuₕ = @. duₕ = hwgrad(hdiv(cuₕ)) # curl-curl term is zero in 1D horizontal
+
+    Spaces.weighted_dss!(dρe)
+    Spaces.weighted_dss!(duₕ)
+
+    κ₄ = 100.0 # m^4/s
+    @. dρe = -κ₄ * hwdiv(cρ * hgrad(χe))
+    @. duₕ = -κ₄ * hwgrad(hdiv(χuₕ)) # curl-curl term is zero in 1D horizontal
```
Comment (szy21) [inline examples/hybrid/bubble3d_invariant_totalenergy.jl:148]: Need to correct this. See #386.

File: examples/hybrid/bubble3d_invariant_totalenergy.jl [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+const grav = 9.8 # gravitational constant
+const R_d = 287.058 # R dry (gas constant / mol mass dry air)
+const γ = 1.4 # heat capacity ratio
+const C_p = R_d * γ / (γ - 1) # heat capacity at constant pressure
+const C_v = R_d / (γ - 1) # heat capacity at constant volume
+const T_0 = 273.16 # triple point temperature
+
+Φ(z) = grav * z
+
+function pressure(ρ, e, uvw, z)
```
Comment (szy21) [inline examples/hybrid/bubble3d_invariant_totalenergy.jl:72]: Need to change this. See [here](https://github.com/CliMA/ClimaCore.jl/blob/0707b365c9f2a539179e1f37524ba9a58ef72b85/examples/3dsphere/baroclinic_wave.jl#L134).

File: examples/hybrid/bubble3d_invariant_totalenergy.jl [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+    cE = @. (norm(cuvw)^2) / 2 + Φ(coords.z)
+    @. duₕ -= hgrad(cE)
+    @. dw -= vgradc2f(cE)
+
+    # 3) potential temperature
+
+    @. dρe -= hdiv(cuvw * cρe)
+    @. dρe -= vdivf2c(fw * Ic2f(cρe))
+    @. dρe -= vdivf2c(Ic2f(cuₕ * cρe))
```
Comment (szy21) [inline examples/hybrid/bubble3d_invariant_totalenergy.jl:242]:
```suggestion
    @. dρe -= hdiv(cuvw * (cρe + cp))
    @. dρe -= vdivf2c(fw * Ic2f(cρe + cp))
    @. dρe -= vdivf2c(Ic2f(cuₕ * (cρe + cp)))
```
see #411

**Comment by akshaysridhar** [comment]
bors try

**Comment by akshaysridhar** [comment]
bors try

**Comment by akshaysridhar** [comment]
bors try

**Pull request merged**